crdschurch/crds-search-index-data
# US19059 Distro update search index

## Task
Only entries with the distribution channel of www.crossroads.net **or** the content model does not have the distribution channel field should be added to the Algolia index. 

The previous implementation removed the records after everything is done. This changes it to remove them before they are sent along with updating to the logic above.

[Rally](https://rally1.rallydev.com/#/66096747656d/custom/24109743995?qdp=%2Fdetail%2Fuserstory%2F371662252388)

## Tester Notes

- **Note:** You will need to do this locally see [Quick Start](https://github.com/crdschurch/crds-search-index-data#quick-start) for setup information.
    - **Tip:** You can set up your own [Algolia](https://www.algolia.com/) instance to make it easier to play around with.
- Create a new article or use this [example article](https://app.contentful.com/spaces/y3a9myzsdjan/environments/int/entries/5NTWz8cailrxUZ49OB4Smo) to toggle **Distribution Channels**.
  - It **should** be added to the index if www.crossroads.net **exists** in the site list
  - It **should not** be added to the index if www.crossroads.net **doesn't exist** in the site list
- Then send a `POST` requests using the following routes.
  - Refresh all Articles in index with `/content/article`. *You can replace article with any content type as long as there is a model for it in `/models/content/content-types`*
  - Refresh all content with `content/refreshAll`
  - Update an individual record using `/content/` with a body containing a contentful content change payload. You can view the search webhook history for int [here](https://app.contentful.com/spaces/y3a9myzsdjan/settings/webhooks/7JxoTxlv99MHxfILw5pgvv) to retrieve an example body.
      - **Note:** I am not sure if it's possible to test an individual article when it's saved on int without it being merged in.
- Check the Algolia index each time by searching using the **ObjectID** instead of the regular **Search**. **Note:** Sometimes the Algolia index search page needed to be refreshed when a record was deleted.
